### PR TITLE
Convert mod_messages to service provider

### PR DESCRIPTION
**Pull request by joomlaweby**
### Summary of Changes

Converts the messages administrator module to service provider.

This is my first pull request to Joomla! project :-).

### Testing Instructions

1. Make sure you have administrator component com_messages (Private Messages) and module mod_messages enabled
2. Make sure you have at least 1 unread message
3. You should see module output

mod_messages

### Actual result BEFORE applying this Pull Request

Module works as expected

### Expected result AFTER applying this Pull Request

Module works same way as before applying pull request

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/modules/mod_messages/src/Dispatcher/Dispatcher.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+     *
+     * @return  void
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    public function dispatch()
+    {
+        // Check permissions.
+        if (!$this->getApplication()->getIdentity()->authorise('core.login.admin')
+            || !$this->getApplication()->getIdentity()->authorise('core.manage', 'com_messages')) {
```
Comment (richard67) [inline administrator/modules/mod_messages/src/Dispatcher/Dispatcher.php]:
```suggestion
        if (
            !$this->getApplication()->getIdentity()->authorise('core.login.admin')
            || !$this->getApplication()->getIdentity()->authorise('core.manage', 'com_messages')
        ) {
```

Comment (richard67) [reply administrator/modules/mod_messages/src/Dispatcher/Dispatcher.php]: The above suggestion should fix the PHPCS check failing in Drone.

File: administrator/modules/mod_messages/src/Helper/MessagesHelper.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+             *  @var \Joomla\Component\Messages\Administrator\Model\MessagesModel $messagesModel
+             *
+             */
+            $messagesModel = $app->bootComponent('com_messages')->getMVCFactory()
+                ->createModel('Messages', 'Administrator', ['ignore_request' => true]);
+            $messagesModel->setState('filter.state', 0);
+            $messages = $messagesModel->getItems();
+
+            return \count($messages);
+        } catch (RuntimeException $e) {
```
Comment (sandewt) [inline administrator/modules/mod_messages/src/Helper/MessagesHelper.php]:
```suggestion
        } catch (\RuntimeException $e) {
```

File: administrator/modules/mod_messages/services/provider.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
+<?php
+
+/**
+ * @package     Joomla.Administrator
+ * @subpackage  mod_messages
+ *
+ * @copyright   (C) 2019 Open Source Matters, Inc. <https://www.joomla.org>
```
Comment (sandewt) [inline administrator/modules/mod_messages/services/provider.php]:
```suggestion
 * @copyright   (C) 2024 Open Source Matters, Inc. <https://www.joomla.org>
```

File: administrator/modules/mod_messages/src/Dispatcher/Dispatcher.php [hunk 4]
```diff
@@ -0,0 +1,63 @@
+<?php
+
+/**
+ * @package     Joomla.Administrator
+ * @subpackage  mod_messages
+ *
+ * @copyright   (C) 2019 Open Source Matters, Inc. <https://www.joomla.org>
```
Comment (sandewt) [inline administrator/modules/mod_messages/src/Dispatcher/Dispatcher.php]:
```suggestion
 * @copyright   (C) 2024 Open Source Matters, Inc. <https://www.joomla.org>
```

File: administrator/modules/mod_messages/src/Helper/MessagesHelper.php (hunk 2 above)
Comment (joomlaweby) [reply administrator/modules/mod_messages/src/Helper/MessagesHelper.php]: Fixed, thanks for feedback.

File: administrator/modules/mod_messages/services/provider.php (hunk 3 above)
Comment (joomlaweby) [reply administrator/modules/mod_messages/services/provider.php]: Fixed

File: administrator/modules/mod_messages/src/Dispatcher/Dispatcher.php (hunk 4 above)
Comment (joomlaweby) [reply administrator/modules/mod_messages/src/Dispatcher/Dispatcher.php]: Fixed, also in MessagesHelper.php file

**Comment by fgsw** [comment]
I have tested this item :white_check_mark: successfully on 24ba818c9029e80a37bcfa919a98b8c2173175f8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42735">issues.joomla.org/tracker/joomla-cms/42735</a>.</sub>

**Comment by richard67** [comment]
@joomlaweby No need to update your branch when there are no conflicts shown. Updating the branch in validates the test counter for human test. I've restored the result from @fgsw now so it's counted again.

**Comment by joomlaweby** [comment]
> @joomlaweby No need to update your branch when there are no conflicts shown. Updating the branch in validates the test counter for human test. I've restored the result from @fgsw now so it's counted again.

> @richard67 OK, got it. :-)

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 769c4abc039e809dbd7ccbc0defc8780f53bfdc0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42735">issues.joomla.org/tracker/joomla-cms/42735</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42735">issues.joomla.org/tracker/joomla-cms/42735</a>.</sub>

**Comment by exlemor** [comment]
Tested successfully during PBF 2024.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42735">issues.joomla.org/tracker/joomla-cms/42735</a>.</sub>

**Comment by LadySolveig** [comment]
Thank you @joomlaweby  and also for testing and support @fgsw @viocassel @exlemor @richard67 @sandewt

**Pull request merged**